infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java

```diff
@@ -421,9 +426,16 @@ private void trimPrefix() {
     }
 
     private LogData getLogData(LogEntry entry) {
-        ByteBuf data = Unpooled.wrappedBuffer(entry.getData().toByteArray());
+        ByteBuffer entryData = ByteBuffer.wrap(entry.getData().toByteArray());
+
+        if (entry.hasCodecType() && entryData.hasRemaining()) {
+            Codec.Type codec = Codec.Type.values()[entry.getCodecType()];
+            entryData = codec.getInstance().decompress(entryData);
+        }
+
         LogData logData = new LogData(org.corfudb.protocols.wireprotocol
-                .DataType.typeMap.get((byte) entry.getDataType().getNumber()), data);
+                .DataType.typeMap.get((byte) entry.getDataType().getNumber()),
+                Unpooled.wrappedBuffer(entryData.array()));
```

Comment:
I understand the necessity to convert between ByteBuf  and ByteBuffer here, but we should add a TODO to later to clean it up.

Reply:
Ostensibly its not good practice to add TODOs, but I have another patch that will eliminate some unnecessary buffer allocations and I'm planning to clean this up.